app/src/main/java/com/a0xffffffff/dinesum/RequestTracker.java

```diff
@@ -4,11 +4,36 @@
 
 public class RequestTracker {
 
+    private static RequestTracker mRequestTracker = new RequestTracker();
+
     private User mCurrentUser;
     private Request[] mNearbyRequests;
 
     private RequestTracker() {
+        mRequestTracker = RequestTracker.getInstance();
+        
+    }
+
+    public static RequestTracker getInstance() {
+        return mRequestTracker;
+    }
+
+    public void filterNearbyRequests(Request[] allRequests) {
```

Comment:
I think it'd be cleaner and safer to return a different Request list, instead of filtering the original Request list

Reply:
You're right, good catch!